js/tiles.js

```diff
@@ -226,6 +226,7 @@ const OpenDoors = {
 
 // door is used in /js/levels.js ignoring the error
 // ignore JSHintBear
+var keyFound=false; //boolean variable to check if key is found or not
```

Comment:
Could you use the inventory for that? @YashKumarVerma can guide you with this.

The problem with global variables is: If we may add a second player in the future, we can not use them any more. So the state needs to be in the player.
I think, most of what you have written there is correct and you can use the inventory to store and ask if the key is there.